src/gatsby/on-render-body.js

```diff
@@ -46,32 +45,32 @@ export default ({ setHeadComponents }, pluginOptions) => {
     // TO DO - Warn about missing options
   }
 
-  let agent;
+  let fetchAgent;
   if (instrumentationType === 'lite') {
-    agent = liteAgent;
+    fetchAgent = fetch("https://js-agent.newrelic.com/nr-loader-rum-current.min.js");
   }
 
   if (instrumentationType === 'pro') {
-    agent = proAgent;
+    fetchAgent = fetch("https://js-agent.newrelic.com/nr-loader-full-current.min.js");
   }
 
   if (instrumentationType === 'proAndSPA') {
-    agent = proAndSpaAgent;
+    fetchAgent = fetch("https://js-agent.newrelic.com/nr-loader-spa-current.min.js");
   }
 
   const configs = `
-    ;NREUM.loader_config={accountID:"${options.accountId}",trustKey:"${options.trustKey}",agentID:"${options.agentID}",licenseKey:"${options.licenseKey}",applicationID:"${options.applicationID}"}
-    ;NREUM.info={beacon:"${options.beacon}",errorBeacon:"${options.errorBeacon}",licenseKey:"${options.licenseKey}",applicationID:"${options.applicationID}",sa:1}
+    ;NREUM.loader_config={accountID:"${options.accountId}",trustKey:"${options.trustKey}",agentID:"${options.agentID}",licenseKey:"${options.licenseKey}",applicationID:"${options.applicationID}"};
+    NREUM.info={beacon:"${options.beacon}",errorBeacon:"${options.errorBeacon}",licenseKey:"${options.licenseKey}",applicationID:"${options.applicationID}",sa:1};
   `;
 
-  if (agent && configs) {
+  fetchAgent?.then(resp => resp.text()).then(agent => {
```

Comment:
I don't fully understand the constraints faced here -- do we need to be cognizant of the `?.` syntax?  We should probably try to be conservative to prevent headaches later.

Reply:
Well aside from safari 10.1-13.1, fetch and optional chain have similar support. Nevermind that the top of the readme for this plugin advertises npm installation, I'm not giving much thought to backwards compatibility for a plugin like this.  🤔